compiler/damlc/lib/DA/Cli/Damlc/Test.hs

```diff
@@ -154,89 +168,200 @@ printSummary color res =
         ]
     printScenarioResults color res
 
+data TemplateIdentifier = TemplateIdentifier
+    { package :: Maybe T.Text -- `package == Nothing` means local package
+    , qualifiedTemplate :: T.Text
+    }
+    deriving (Eq, Ord, Show)
+
+data ChoiceIdentifier = ChoiceIdentifier
+    { packageTemplate :: TemplateIdentifier
+    , choice :: T.Text
+    }
+    deriving (Eq, Ord, Show)
+
+type TemplateInfo = (Maybe LF.PackageId, LF.Module, LF.Template)
```

Comment:
I'd suggest using [`DA.Daml.LF.Ast.Base.Qualified`](https://github.com/digital-asset/daml/blob/main/compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Base.hs#L135-L141)` LF.Template` instead to avoid the 3-tuple and the `Maybe` field